Modules/LearningSequence/classes/Player/class.ilLegacyKioskModeView.php

```diff
@@ -68,12 +68,29 @@ public function buildInitialState(State $empty_state): State
 	 */
 	public function buildControls(State $state, ControlBuilder $builder): ControlBuilder
 	{
-		$builder->start (
```

Comment:
Technically this did not work in the first place (and PHPStorm should have complained), because `ControlBuilder` has no `start`-method (only `LSControlBuilder` has). In fact, this is never used with any other `ControlBuilder` atm, so this never became a problem. Same goes for the new methods that are exclusive to `LSControlBuilder`.

Please add a guard clause that checks if the `$builder` is indeed an `LSControlBuilder` and throws a `LogicException` otherwise.